src/Graphing.hs

```diff
@@ -160,6 +161,15 @@ stripRoot (Graphing gr) = Graphing $ AM.overlay newDirectEdges (AM.removeVertex
     newDirectEdges :: AM.AdjacencyMap (Node ty)
     newDirectEdges = AM.edges $ map (Root,) newDirect
 
+-- | Shrinks all root nodes.
+-- Remove current direct nodes. It will promote their immediate children as directs.
+-- Unlike @stripRoot@, it removes them from graphing, instead of preserving them (as node), and their edges.
```

Comment:
I bet this is the intended behavior for `stripRoot`, and it's just implemented wrong. It's probably another case where we implicitly expect `pruneUnreachable` to remove those nodes after `stripRoot`

We should probably replace the existing `stripRoot` with this implementation

Reply:
Agreed - right now, yarnlock, and cargo is using this strategy. I have added tech-debt ticket in backlog for this https://github.com/fossas/team-analysis/issues/746 (prefer to keep the scope/blast-zone of this PR only related to mvn).

Reply:
I'm concerned that by not doing it now we'll never get around to it, unless you plan on doing it as an immediate followup to this

It'll mean one fewer thing for us to worry about when dealing with the tech debt of removing the implicit `pruneUnreachable`

Reply:
Yeah - I plan to have followup PR this week for this. I've assigned the ticket it to myself.